pkgs/tools/misc/dtool/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "dtool";
+  version = "0.12.0";
+
+  src = fetchFromGitHub {
+    owner = "guoxbin";
+    repo = "dtool";
+    rev = "v${version}";
+    hash = "sha256-m4H+ANwEbK6vGW3oIVZqnqvMiAKxNJf2TLIGh/G6AU4=";
+  };
+
+  cargoHash = "sha256-r8r3f4yKMQgjtB3j4qE7cqQL18nIqAGPO5RsFErqh2c=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  meta = with lib; {
+    description = "A command-line tool collection to assist development written in RUST";
+    homepage = "https://github.com/guoxbin/dtool";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ ];
```

Comment:
you should add your self as a maintainer. If you don't already have a maintainer entry add one in https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix and make it it's on commit called `maintainers: add yourName`

Reply:
I've added myself